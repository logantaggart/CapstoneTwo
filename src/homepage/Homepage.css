.homepage {
    background-color: rgb(81, 118, 230);
}

.logo {
    color: seashell;
    font-size: 75px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    padding-bottom: 1.5em;
}

.description {
    font-size: 1.75em;
    padding-bottom: 1.25em;
}

.linkbtn {
    margin-left: 15px;
}