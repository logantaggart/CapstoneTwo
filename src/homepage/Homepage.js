import React, { useState, useEffect, useContext } from 'react'
import { Link } from 'react-router-dom'

import UserContext from '../auth/UserContext'

import CryptoApi from '../api/api'

import Currency from '../currency/Currency'

import './Homepage.css'


function Homepage(props) {
    const { currentUser } = useContext(UserContext)

    const [formData, setFormData] = useState({
        cryptoname: "",
        convertname: "USD",
        isSubmitted: false
    })

    const [cryptoInfo, setCryptoInfo] = useState({
        information: "",
        conversionCurr: "USD"
    })

    useEffect(() => {
        console.log("Changed:", cryptoInfo)
    }, [cryptoInfo])

    useEffect(() => {
        console.log("Changed:", formData)
    }, [formData.isSubmitted])

    async function handleSubmit(evt) {
        evt.preventDefault()

        let info = await CryptoApi.getCoin({ cryptoname: formData.cryptoname.toLowerCase(), convertname: formData.convertname })

        setCryptoInfo({ information: info.data, conversionCurr: formData.convertname })
        setFormData({ cryptoname: "", convertname: "USD", isSubmitted: true })
    }

    function handleChange(evt) {
        const { name, value } = evt.target
        setFormData(s => ({ ...s, [name]: value }))
    }


    return (
        <div className="homepage">
            <div className="container text-center">
                <h1 className="mb-3 font-weight-bold logo">Cryptic</h1>
                {currentUser ?
                    <div>
                        <form onSubmit={handleSubmit}>
                            <input
                                name="cryptoname"
                                placeholder="Please Enter A Cryptocurrency Name (Ex: Ethereum)"
                                className="form-control form-control-lg flex-grow-1"
                                onChange={handleChange}
                                value={formData.cryptoname}
                                required
                            />

                            <br />

                            <select name="convertname" onChange={handleChange} value={formData.convertname} className="form-control form-control-sm flex-grow-1">
                                <option value="USD">USD ($)</option>
                                <option value="EUR">EUR (€)</option>
                                <option value="JPY">JPY (¥)</option>
                                <option value="GPY">GBP (£)</option>
                                <option value="AUD">AUD (A$)</option>
                                <option value="CAD">CAD (C$)</option>
                                <option value="CHF">CHF (Fr)</option>
                                <option value="CNY">CNY (¥)</option>
                                <option value="HKD">HKD (HK$)</option>
                                <option value="NZD">NZD (NZ$)</option>
                                <option value="SEK">SEK (Kr)</option>
                                <option value="KRW">KRW (₩)</option>
                                <option value="SGD">SGD (S$)</option>
                                <option value="NOK">NOK (Kr)</option>
                                <option value="MXN">MXN ($)</option>
                                <option value="INR">INR (₹)</option>
                                <option value="RUB">RUB (₽)</option>
                                <option value="ZAR">ZAR (R)</option>
                                <option value="TRY">TRY (₺)</option>
                                <option value="BRL">BRL (R$)</option>
                            </select>
                        </form>
                    </div>
                    : <>
                        <p className="description">Cryptic is a tool built for the future of finance that allows users to find and save information on their cryptocurrency of choice.</p>
                        <Link className="btn btn-primary btn-lg font-weight-bold mr-3 linkbtn" to="/login">
                            Log in
                        </Link>
                        <Link className="btn btn-primary btn-lg font-weight-bold linkbtn" to="/signup">
                            Sign up
                        </Link>
                    </>}

                {currentUser && formData.isSubmitted ? <Currency data={cryptoInfo} retrieveWatchlist={props.retrieveWatchlist} /> : null}
            </div>
        </div>
    )
}

export default Homepage