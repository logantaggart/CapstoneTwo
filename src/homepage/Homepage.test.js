import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import UserProvider from '../testUtils'

import Homepage from './Homepage'

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Homepage />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when logged out', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <Homepage />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
})