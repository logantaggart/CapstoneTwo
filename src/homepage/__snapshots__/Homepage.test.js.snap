// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-3 font-weight-bold logo"
      >
        Cryptic
      </h1>
      <div>
        <form>
          <input
            class="form-control form-control-lg flex-grow-1"
            name="cryptoname"
            placeholder="Please Enter A Cryptocurrency Name (Ex: Ethereum)"
            required=""
            value=""
          />
          <br />
          <select
            class="form-control form-control-sm flex-grow-1"
            name="convertname"
          >
            <option
              value="USD"
            >
              USD ($)
            </option>
            <option
              value="EUR"
            >
              EUR (€)
            </option>
            <option
              value="JPY"
            >
              JPY (¥)
            </option>
            <option
              value="GPY"
            >
              GBP (£)
            </option>
            <option
              value="AUD"
            >
              AUD (A$)
            </option>
            <option
              value="CAD"
            >
              CAD (C$)
            </option>
            <option
              value="CHF"
            >
              CHF (Fr)
            </option>
            <option
              value="CNY"
            >
              CNY (¥)
            </option>
            <option
              value="HKD"
            >
              HKD (HK$)
            </option>
            <option
              value="NZD"
            >
              NZD (NZ$)
            </option>
            <option
              value="SEK"
            >
              SEK (Kr)
            </option>
            <option
              value="KRW"
            >
              KRW (₩)
            </option>
            <option
              value="SGD"
            >
              SGD (S$)
            </option>
            <option
              value="NOK"
            >
              NOK (Kr)
            </option>
            <option
              value="MXN"
            >
              MXN ($)
            </option>
            <option
              value="INR"
            >
              INR (₹)
            </option>
            <option
              value="RUB"
            >
              RUB (₽)
            </option>
            <option
              value="ZAR"
            >
              ZAR (R)
            </option>
            <option
              value="TRY"
            >
              TRY (₺)
            </option>
            <option
              value="BRL"
            >
              BRL (R$)
            </option>
          </select>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-3 font-weight-bold logo"
      >
        Cryptic
      </h1>
      <p
        class="description"
      >
        Cryptic is a tool built for the future of finance that allows users to find and save information on their cryptocurrency of choice.
      </p>
      <a
        class="btn btn-primary btn-lg font-weight-bold mr-3 linkbtn"
        href="/login"
      >
        Log in
      </a>
      <a
        class="btn btn-primary btn-lg font-weight-bold linkbtn"
        href="/signup"
      >
        Sign up
      </a>
    </div>
  </div>
</DocumentFragment>
`;
